geode-experimental-driver/src/main/java/org/apache/geode/experimental/driver/ProtobufRegion.java

```diff
@@ -153,4 +155,18 @@ public void remove(K key) throws IOException {
 
     protobufChannel.sendRequest(request, MessageTypeCase.REMOVERESPONSE);
   }
+
+  @Override
+  public Set<K> keySet() throws IOException {
+    final Message request = Message.newBuilder()
+        .setKeySetRequest(RegionAPI.KeySetRequest.newBuilder().setRegionName(name)).build();
+    final Message message = protobufChannel.sendRequest(request, MessageTypeCase.KEYSETRESPONSE);
+    final RegionAPI.KeySetResponse keySetResponse = message.getKeySetResponse();
+
+    Set<K> keys = new HashSet<>(keySetResponse.getKeysCount());
+    for (BasicTypes.EncodedValue value : keySetResponse.getKeysList()) {
```

Comment:
I keep wondering if there's some lambda way to do this kind of thing...

Reply:
Yeah, I could certainly stream them through a map and then collect them.  This approach just seemed a bit clearer though, in case someone was using this driver as a reference.